### PR TITLE
Various improvements and bugfixes

### PR DESCRIPTION
**Pull request by Jarpletz**
Improvements to batting and pitching tables
- Made Player's name a link to their baseball reference page
- Hide players who have 0 plate appearences

Improvements to immaculate grid
- Displays the year in which the player completed a question (because sometimes there are multiple players with the same name at different times, so you know which to pick when entering it into immaculategrid.com)
   - grid solver is now returning a touple instead of a string
- Was having bugs with people doing something across their whole season, but not all on the requested team
   - added grouping by team to some queries that were missing it but needed it 
   - likewise, joining on year when you merge the queries together to double ensure an event occurs while the player was on the specified team
 - The "World Series Champ" question was not working right, so fixed that too

### CODE REVIEW

File: immaculateGridCalculations/gridSolver.py [hunk 1]
```diff
@@ -255,7 +265,8 @@ def getPlayerSeasonK(minK):
     query = (
         db.session.query(
             Pitching.playerID,
-            Pitching.teamID.label("teamID")
+            Pitching.teamID.label("teamID"),
+            Batting.yearID.label("yearID")
         )
         .group_by(Pitching.playerID, Pitching.yearID, Pitching.teamID)  # Group by playerID, yearId, and teamID
```
Comment (Renzenar) [inline immaculateGridCalculations/gridSolver.py:297]: Wouldn't Pitching.yearID need to be replaced by Batting.yearID in the group_by statement?

File: README.md
Comment (Renzenar) [inline README.md:1]: I like the personality here! I feel like the links to the player's pages on baseball reference deserves a mentions.

**Review by Renzenar — CHANGES_REQUESTED** [review]
Just the one question and suggestion. Otherwise looks good!

File: immaculateGridCalculations/gridSolver.py (hunk 1 above)
Comment (Jarpletz) [reply immaculateGridCalculations/gridSolver.py:297]: Ahh, good catch, I missed that while copy pasting!

File: README.md
Comment (Jarpletz) [reply README.md:1]: ooh yess thanks

**Comment by Jarpletz** [comment]
Added a few changes:
1. Fixed your comments
2. Updated readme as you suggested
3. Grid Solver can no longer return the same player twice

**Review by Renzenar — APPROVED** [review]
Good

**Pull request merged**